app.rb

```diff
@@ -0,0 +1,101 @@
+require "sinatra/base"
+require "sinatra/flash"
+require "./lib/user.rb"
+require "./lib/message.rb"
+require_relative "./database_connection_setup.rb"
+
+class Chitter < Sinatra::Base
+  enable :sessions, :method_override
+  register Sinatra::Flash
+
+  get "/message" do
+    @messages = Message.all
+    @messages = nil if !(@messages.is_a? Array)
+    erb :"message/message"
+  end
+
+  post "/message/new" do
+    @message = Message.create(text: params[:message], sender: session[:userid])
+    flash[:messagenotice] = "New message created"
+    redirect "/message"
+  end
+
+  delete '/message/:id' do
+    Message.delete(id: params[:id])
+    flash[:messagenotice] = "Message deleted"
+    redirect '/message'
+  end
+
+  patch "/message/:id" do
+    @message = Message.update(text: params[:updatetext], sender: session[:userid],id: params[:id])
+    flash[:messagenotice] = "Message updated"
+    redirect "/message"
+  end
+
+  post "/message/:id/reply" do
+    @message = Message.reply(text: params[:replytext], sender: session[:userid], response_to: params[:id])
+    flash[:messagenotice] = "Message replied to"
+    redirect "/message"
+  end
+
+  get "/message/:id/reply" do
+    @message = Message.find(id: params[:id])
+    erb :"message/reply"
+  end
+
+  get "/message/:id/update" do
+    @message = Message.find(id: params[:id])
+    erb :"message/update"
+  end
+
+  get "/user" do
+    erb :"user/login"
+  end
+
+  get "/user/logout" do
+    session.clear
+    flash[:usernotice] = "Logout successful"
+    redirect "/message"
+  end
+
+  post "/user" do
+    user = User.login(name: params[:name],password: params[:password])
+    if user.is_a?(User)
```

Comment:
The idea you're applying here of determining which route an application should follow based on the result of a db query is a very common one.

More conventionally, you would simply have your login method return false or nil if it fails allowing you to do something like

```ruby
user = User.login(name: params[:name],password: params[:password])
if user 
  //routing logic
else 
  //other routing logic
end
```